src/listing.rs

```diff
@@ -260,3 +306,39 @@ pub fn directory_listing<S>(
         .content_type("text/html; charset=utf-8")
         .body(html))
 }
+
+/// Converts a SystemTime object to a strings tuple (date, time)
+/// Date is formatted as %e %b, e.g. Jul 12
+/// Time is formatted as %R, e.g. 22:34
+///
+/// If no SystemTime was given, returns a tuple containing empty strings
+fn convert_to_utc(src_time: Option<SystemTime>) -> (String, String) {
+    match src_time {
+        Some(time) => {
+            let date_time = DateTime::<Utc>::from(time);
+            (
+                date_time.format("%e %b").to_string(),
+                date_time.format("%R").to_string(),
+            )
+        }
+        None => ("".to_string(), "".to_string()),
+    }
+}
+
+/// Converts a SystemTime to a string readable by a human,
+/// i.e. calculates the duration between now() and the given SystemTime,
+/// and gives a rough approximation of the elapsed time since
+///
+/// If no SystemTime was given, returns an empty string
+fn humanize_duration(src_time: Option<SystemTime>) -> String {
```

Comment:
Wrong name, I think, as this is not a duration but a point in time. So humanize_systemtime?

On a side note, wouldn't it be cool actually put a function like this into chrono_humanize itself?

Reply:
True, that would be a good addition to the crate